pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -341,13 +356,19 @@ func (r *ReconcileConstraintTemplate) handleUpdate(
 			log.Error(err, "failed to report constraint template ingestion duration")
 		}
 		err := r.reportErrorOnCTStatus("ingest_error", "Could not ingest Rego", ct, err)
+		r.tracker.CancelTemplate(unversionedCT) // Don't track templates that failed compilation
```

Comment:
I'm wondering if we should have an error count threshold here. We don't want to hard-down the webhook on a bad template, but we also don't want to relax the warming requirement due to transient errors. Keeping a counter inside of tracker that expires a resource after a certain threshold seems like it may strike the right balance.

Reply:
Let's address as a followup - it's hard to differentiate between transient and non-transient errors, let's continue to explore what the right balance will be here.

Reply:
Can you pls create a followup issue for this?

Reply:
https://github.com/kubernetes-sigs/controller-runtime/issues/948